movies/1.entry-point.md

```diff
@@ -0,0 +1,67 @@
+# API entrypoint
+
+## Story
+
+As an API client I want to start with some URI to be able to discover the API capabilities.
+
+## API behaviour
+
+```
+GET https://hydra-movies.herokuapp.com/
+```
+
+Response headers (excerpt):
+
+```
+Access-Control-Allow-Origin: *
+Access-Control-Expose-Headers: Location, Link
+Link: </doc>; rel="http://www.w3.org/ns/hydra/core#apiDocumentation"
+```
+
+Response body (excerpt):
+
+```json
+{
+  "@context": "/context.jsonld",
+  "@id": "/",
+  "@type": "schema:EntryPoint"
+}
+```
+### Explanation
+
+```
+GET https://hydra-movies.herokuapp.com/
+```
+
+The example API is hosted at https://hydra-movies.herokuapp.com/, so most commonly this is the URI we will give to clients initially. Clients that use the API for the first time will start by dereferencing this URI.
+
+```
+Access-Control-Allow-Origin: *
+Access-Control-Expose-Headers: Location, Link
+```
+
+As the API will be used by a variety of clients, we need to ensure a proper setup for Cross Origin Resource Sharing (CORS). The example API allows access from any Origin and exposes some relevant headers. Exposing the `Link` header is a **must have** for Hydra APIs, because the API documentation is discovered this way.
+
+```
+Link: </doc>; rel="http://www.w3.org/ns/hydra/core#apiDocumentation"
+```
+
+Hydra uses the HTTP `Link` header to refer to the API documentation. `</doc>` is a relative Link to the documentation and `rel="http://www.w3.org/ns/hydra/core#apiDocumentation"` is a fixed value that describes the type of the link. Every API resource (not only the entrypoint) must provide this header with every response. The client knows now, that the API documentation can be found at https://hydra-movies.herokuapp.com/doc.
```

Comment:
Worth noting that failed requests should include the link too?

Reply:
yep